### ISSUE TITLE
RFC2445 filter running out of memory on Unicode text

### ISSUE THREAD

**Issue by rungta** [nystudio107/craft-icalendar#34]
Hi Andrew 👋!

### Describe the bug
Might have encountered a memory leak. PHP is running out of memory when applying the `rfc2445` filter on some Unicode content (in Devanagari) once it exceeds a certain length/threshold.

### To reproduce
Steps to reproduce the behaviour:
1. Run the following Twig code
    ```twig
    {% apply rfc2445 %}
      BEGIN: VEVENT
      DESCRIPTION: कोरोना के चलते बच्चों की शिक्षा का बहुत नुकसान हुआ है। अब जबकि स्कूल खुल गये हैं तो प्राथमिक जरूरत है बच्चे क्या भूले हैं और उन्हें क्या याद है इसे ठीक तरह से जानना और उनके मन को सहेजते हुए उनके साथ सीखने-सिखाने की ओर बहुत तसल्ली से धैर्य के साथ बढ़ना ।
      END: VEVENT
    {% endapply %}
    ```
2. See error message
   ```
   PHP Fatal Error – [yii\base\ErrorException](http://www.yiiframework.com/doc-2.0/yii-base-errorexception.html)
   Allowed memory size of 134217728 bytes exhausted (tried to allocate 67108872 bytes)
   
   yii\base\ErrorException: Allowed memory size of 134217728 bytes exhausted (tried to allocate 67108872 bytes) in ./vendor/nystudio107/craft-icalendar/src/services/Convert.php:91
   Stack trace:
   #0 [internal function]: yii\base\ErrorHandler->handleFatalError()
   #1 {main}
   ```

### Expected behaviour
The plugin should be able to process this within the allocated 128 MBs.

### Versions
- PHP version: 7.4.30
- Plugin version: 1.1.3
- Craft version: 3.7.47.1

**Comment by khalwat** [comment]
@rungta can you upload a .ics file that you're using, so I can test it locally?

**Comment by rungta** [comment]
Here's an `.ics` file that was successfully generated:

```BEGIN:VCALENDAR
VERSION:2.0
CALSCALE:GREGORIAN
METHOD:PUBLISH
PRODID:-//Azim Premji University//Events//EN
BEGIN:VEVENT
DTSTART:20220812
DTEND:20220814
UID:8096cd73-b822-4f20-a1d0-2e503f598b48
SUMMARY:Campus Week 2022
LOCATION:Azim Premji University\, Bengaluru
URL:https://azimpremjiuniversity.edu.in/events/2022/campus-week-2022
DTSTAMP:20220805T104600Z
DESCRIPTION:From crochet for a cause to painting a murder scene, there are 
	more than 90 events in store at our very first Campus Week.
END:VEVENT
END:VCALENDAR
```

But the same thing fails when the `DESCRIPTION` contains Devanagari content. You can see it in action on these pages:
- [Working example](https://azimpremjiuniversity.edu.in/events/2022/campus-week-2022)
- [Bug](https://azimpremjiuniversity.edu.in/events/2022/%E0%A4%B6%E0%A5%88%E0%A4%95%E0%A5%8D%E0%A4%B7%E0%A4%BF%E0%A4%95-%E0%A4%AA%E0%A5%8D%E0%A4%B0%E0%A4%B5%E0%A4%BE%E0%A4%B9-%E0%A4%AC%E0%A4%BE%E0%A4%B2-%E0%A4%B8%E0%A4%BE%E0%A4%B9%E0%A4%BF%E0%A4%A4%E0%A5%8D%E0%A4%AF-%E0%A4%B8%E0%A5%87-%E0%A4%AA%E0%A5%9D%E0%A4%A8%E0%A4%BE-%E0%A4%B8%E0%A5%80%E0%A4%96%E0%A4%A8%E0%A4%BE)

**Comment by khalwat** [comment]
Hooked it up to XDebug, and figured it out @rungta -- basically, it wasn't using `mb_` variants of things like `strlen()` so it was failing on multi-byte encoded content.

So this:

```twig
    {% apply rfc2445 %}
        BEGIN: VEVENT
        DESCRIPTION: कोरोना के चलते बच्चों की शिक्षा का बहुत नुकसान हुआ है। अब जबकि स्कूल खुल गये हैं तो प्राथमिक जरूरत है बच्चे क्या भूले हैं और उन्हें क्या याद है इसे ठीक तरह से जानना और उनके मन को सहेजते हुए उनके साथ सीखने-सिखाने की ओर बहुत तसल्ली से धैर्य के साथ बढ़ना ।
        END: VEVENT
    {% endapply %}
```

...now generates this:
```
BEGIN: VEVENT
DESCRIPTION: कोरोना के चलते बच्चों की शिक्षा का बहुत नुकसान हुआ है। अब जबकि
	 स्कूल खुल गये हैं तो प्राथमिक जरूरत है बच्चे क्या भूले हैं और उन्हें क्या
	 याद है इसे ठीक तरह से जानना और उनके मन को सहेजते हुए उनके साथ सीखने-सिखान
	े की ओर बहुत तसल्ली से धैर्य के साथ बढ़ना ।
END: VEVENT
```

Addressed in: https://github.com/nystudio107/craft-icalendar/commit/e3ec1be1364bd354eb44ca9ae0437529835f610f & https://github.com/nystudio107/craft-icalendar/commit/73d954ea36aa6f71b29381dab1cf27a53fa7d129

Craft CMS 3:

You can try it now by setting your semver in your `composer.json` to look like this:
```json
    "nystudio107/craft-icalendar": "dev-develop as 1.1.4”,
```
Then do a `composer clear-cache && composer update`


Craft CMS 4:

…..

You can try it now by setting your semver in your `composer.json` to look like this:
```json
    "nystudio107/craft-icalendar": "dev-develop-v4 as 4.0.1”,
```
Then do a `composer clear-cache && composer update`

If you can try it out and verify that it works for your use-case, I'll cut a release @rungta

**Comment by rungta** [comment]
Works like a charm @khalwat! Thanks 🙌

(Tested the Craft 3 version)

**Comment by khalwat** [comment]
Released as 1.1.4 https://github.com/nystudio107/craft-icalendar/releases/tag/1.1.4 & 4.0.1 https://github.com/nystudio107/craft-icalendar/releases/tag/4.0.1
